src/main/java/org/folio/circulation/resources/CheckOutByBarcodeResource.java

```diff
@@ -208,33 +163,69 @@ private String urlForLoan(String id) {
-    Result<User> getUserResult) {
+    Result<User> getUserResult, CirculationErrorHandler errorHandler) {
+
+    if (getUserResult.value() == null || errorHandler.hasAny(FAILED_TO_FETCH_USER)) {
+      return loanResult;
+    }
 
-    return loanResult.combine(getUserResult, LoanAndRelatedRecords::withRequestingUser);
+    return loanResult.combine(getUserResult, LoanAndRelatedRecords::withRequestingUser)
+      .mapFailure(failure -> errorHandler.handleValidationError(failure, FAILED_TO_FETCH_USER, loanResult));
+  }
+
+  private CompletableFuture<Result<User>> getProxyUserByBarcode(String barcode,
+    UserRepository userRepository, CirculationErrorHandler errorHandler) {
+
+    return userRepository.getProxyUserByBarcode(barcode)
+      .thenApply(r -> errorHandler.handleValidationResult(r, FAILED_TO_FETCH_PROXY_USER, succeeded(null)));
+  }
+
+  private Result<LoanAndRelatedRecords> addProxyUser(Result<LoanAndRelatedRecords> loanResult,
+    Result<User> getUserResult, CirculationErrorHandler errorHandler) {
+
+    if (getUserResult.value() == null ||
+      errorHandler.hasAny(FAILED_TO_FETCH_PROXY_USER)) {
+
+      return loanResult;
+    }
+
+    return loanResult.combine(getUserResult, LoanAndRelatedRecords::withProxyingUser)
+      .mapFailure(failure -> errorHandler.handleValidationError(failure, FAILED_TO_FETCH_PROXY_USER, loanResult));
+  }
+
+  private CompletableFuture<Result<Item>> getItemByBarcode(String barcode,
+    ItemRepository itemRepository, CirculationErrorHandler errorHandler) {
+
+    return itemRepository.fetchByBarcode(barcode)
+      .thenApply(r -> errorHandler.handleValidationResult(r, FAILED_TO_FETCH_ITEM, succeeded(null)));
   }
 
   private Result<LoanAndRelatedRecords> addItem(Result<LoanAndRelatedRecords> loanResult,
-    Result<Item> inventoryRecordsResult) {
+    Result<Item> inventoryRecordsResult, CirculationErrorHandler errorHandler) {
+
+    if (inventoryRecordsResult.value() == null ||
+      errorHandler.hasAny(FAILED_TO_FETCH_ITEM)) {
+
+      return loanResult;
+    }
 
     return loanResult.combine(inventoryRecordsResult, LoanAndRelatedRecords::withItem);
   }
 
-  private static ValidationErrorFailure errorWhenInIncorrectStatus(Item item) {
-    String message =
-      String.format("%s (%s) (Barcode: %s) has the item status %s and cannot be checked out",
-        item.getTitle(),
-        item.getMaterialTypeName(),
-        item.getBarcode(),
-        item.getStatusName());
+  private CompletableFuture<Result<LoanAndRelatedRecords>> getRequestQueue(
+    LoanAndRelatedRecords loanAndRelatedRecords, RequestQueueRepository requestQueueRepository,
+    CirculationErrorHandler errorHandler) {
 
-    return singleValidationError(message, ITEM_BARCODE, item.getBarcode());
+    return requestQueueRepository.get(loanAndRelatedRecords)
+      .thenApply(r -> errorHandler.handleValidationResult(r, FAILED_TO_FETCH_REQUEST_QUEUE, loanAndRelatedRecords));
```

Comment:
`requestQueueRepository.get(loanAndRelatedRecords)` does no seem to throw validation errors.